src/System.Security.Cryptography.Pkcs/src/Internal/Cryptography/Pal/AnyOS/PkcsPal.AnyOS.cs

```diff
@@ -0,0 +1,217 @@
+                out int version,
+                out ContentInfo contentInfo,
+                out AlgorithmIdentifier contentEncryptionAlgorithm,
+                out X509Certificate2Collection originatorCerts,
+                out CryptographicAttributeObjectCollection unprotectedAttributes)
+            {
+                throw new PlatformNotSupportedException(SR.PlatformNotSupported_CryptographyPkcs);
+            }
+
+            public override byte[] EncodeOctetString(byte[] octets)
+            {
+                // Write using DER to support the most readers.
+                AsnWriter writer = new AsnWriter(AsnEncodingRules.DER);
+                writer.WriteOctetString(octets);
+                return writer.Encode();
+            }
+
+            public override byte[] DecodeOctetString(byte[] encodedOctets)
+            {
+                // Read using BER because the CMS specification says the encoding is BER.
+                AsnReader reader = new AsnReader(encodedOctets, AsnEncodingRules.BER);
+
+                const int ArbitraryStackLimit = 256;
+                Span<byte> tmp = stackalloc byte[ArbitraryStackLimit];
+                ReadOnlySpan<byte> data = stackalloc byte[0];
+                byte[] poolBytes = null;
+
+                try
+                {
+                    if (!reader.TryGetPrimitiveOctetStringBytes(out var contents))
+                    {
+                        if (reader.TryCopyOctetStringBytes(tmp, out int bytesWritten))
+                        {
+                            data = tmp.Slice(0, bytesWritten);
+                        }
+                        else
+                        {
+                            poolBytes = ArrayPool<byte>.Shared.Rent(reader.PeekContentBytes().Length);
+
+                            if (!reader.TryCopyOctetStringBytes(poolBytes, out bytesWritten))
+                            {
+                                Debug.Fail("TryCopyOctetStringBytes failed with a provably-large-enough buffer");
+                                throw new CryptographicException();
+                            }
+
+                            data = tmp.Slice(0, bytesWritten);
+                        }
+                    }
+                    else
+                    {
+                        data = contents.Span;
+                    }
+
+                    if (reader.HasData)
+                    {
+                        throw new CryptographicException(SR.Cryptography_Der_Invalid_Encoding);
+                    }
+
+                    return data.ToArray();
+                }
+                finally
+                {
+                    if (poolBytes != null)
+                    {
+                        Array.Clear(poolBytes, 0, data.Length);
+                        ArrayPool<byte>.Shared.Return(poolBytes);
+                    }
+                }
+            }
+
+            public override byte[] EncodeUtcTime(DateTime utcTime)
```

Comment:
I was going to say "EnvelopedCms was added in NetFx 1.1 and DateTimeOffset was added in 3.5", but they were both 2.0.  So... no idea why DateTime is what ended up in the public API.

For this method, it's just because the PAL layer already defined it that way and I just made it work.